### PR TITLE
Set default parameter location based on consumes

### PR DESCRIPTION
**Pull request by dhruvCW**
Just a fork of #880 that is based on the latest changes in master.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -75,3 +75,6 @@ Style/RegexpLiteral:
 
 Style/SlicingWithRange:
   Enabled: false
+
+Metrics/ParameterLists:
```
Comment (LeFnord) [inline .rubocop.yml:79]: please remove it here, and add `# rubucop:disable Metrics/ParameterLists` before the method and after `# rubucop:enable Metrics/ParameterLists`

File: lib/grape-swagger/doc_methods/parse_params.rb [hunk 2]
```diff
@@ -148,14 +148,18 @@ def document_example(settings)
...
-        def param_type(value_type)
+        def param_type(value_type, consumes)
           param_type = value_type[:param_type] || value_type[:in]
           if value_type[:path].include?("{#{value_type[:param_name]}}")
             'path'
           elsif param_type
             param_type
           elsif %w[POST PUT PATCH].include?(value_type[:method])
-            DataType.request_primitive?(value_type[:data_type]) ? 'formData' : 'body'
+            if consumes.include?('application/x-www-form-urlencoded') || consumes.include?('multipart/form-data')
```
Comment (LeFnord) [inline lib/grape-swagger/doc_methods/parse_params.rb:158]: what is if `consumes` isn't defined?

**Review by LeFnord — COMMENTED** [review]
thanks @dhruvCW …
but I have one big question, ite seems now the consumes key must be given,
is ot what you want? 
want to see a fallback, if it isn't defined then json should be consumed

File: lib/grape-swagger/doc_methods/parse_params.rb (hunk 2 above)
Comment (dhruvCW) [reply lib/grape-swagger/doc_methods/parse_params.rb:158]: the method expects `consumes` to be an array of consumed types i.e. `application/json` or `application/x-www-form-urlencoded`.

consumes is defined by [on this line](https://github.com/ruby-grape/grape-swagger/blob/master/lib/grape-swagger/endpoint.rb#L121) and by default (if consumes isn't defined) is set to `application/json`.

This PR basically changes the default value of `in` field to `body` unless the consumes value is explicitly set to `urlencoded`.

Would you like me to define a default value for consumes in the `call` method as well ?

**Review by LeFnord — APPROVED** [review]
yeah @dhruvCW it makes sense, thanks

**Pull request merged**